langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/Filter.java

```diff
@@ -44,7 +44,7 @@
                                                                       ARRAY_VERSION, "filter");
 
     public static BArray filter(BArray arr, BFunctionPointer<Object, Boolean> func) {
```

Comment:
Do we have set of methods we need to support read only?. I can see array reverse also we create array value with given array type without using element type. @MaryamZi @HindujaB

Reply:
The doc does not specifically say that we return a new array (unlike some other functions as filter, map etc.). 

So I found that currently we are not allowing the usage of `reverse` functionality for `readonly` types from front-end. 
https://ballerina.io/swan-lake/learn/api-docs/ballerina/lang.array/functions.html#reverse

Reply:
With reverse I think there's actually a spec deviation, we shouldn't be returning a new array. See https://github.com/ballerina-platform/ballerina-lang/issues/26843.